src/chat/chatter.py

```diff
@@ -271,20 +274,26 @@ def updatePlayer(self):
     def update_country(self):
         player = self.user_player
         if player is None:
+            if self.toolTip() == "":  # EARLY OUT
+                return
             self.setIcon(QtGui.QIcon())
             self.setToolTip("")
             return
         # server sends '' for no ip2country-resolution
         if player.country is None or player.country == '':
             country = '__'
         else:
-            country = player.country.lower()
-        self.setIcon(util.THEME.icon("chat/countries/{}.png".format(country)))
+            country = player.country
+        if self.toolTip() == country:  # EARLY OUT
```

Comment:
I assume it's here to avoid a call to setIcon, but is there a measurable performance impact to not avoiding the call?

Reply:
may I quote myself: "speedup (50% faster on average)"

Reply:
50% is relative. Is updating chatters a performance bottleneck in any way?

Reply:
This still seems unnecessary.